src/tiramisu_core.cpp

```diff
@@ -794,13 +804,13 @@ void tiramisu::computation::separate(int dim, tiramisu::expr N, int v)
     constraint += "]: ";
 
     std::string constraint1 = constraint +
-                                this->get_dimension_name_for_loop_level(dim) + " < (" + std::to_string(v) + "*(floor((" + N_without_cast + ")/" + std::to_string(v) + ")))}";
+                        this->get_dimension_name_for_loop_level(dim) + " < (" + std::to_string(v) + "*( floor((" + N_without_cast + ")/" + std::to_string(v) + ")) +"+lower_without_cast+" )}";
     DEBUG(3, tiramisu::str_dump("The constraint is:" + constraint1));
 
     // We create the constraint (i >= v*floor(N/v))
     DEBUG(3, tiramisu::str_dump("Constructing the constraint (i>=v*(floor(N/v)))"));
     std::string constraint2 = constraint +
-                                this->get_dimension_name_for_loop_level(dim) + " >= (" + std::to_string(v) + "*(floor((" + N_without_cast + ")/" + std::to_string(v) + ")))}";
+                    this->get_dimension_name_for_loop_level(dim) + " >= (" + std::to_string(v) + "*(floor((" + N_without_cast + ")/" + std::to_string(v) + ")) +"+lower_without_cast+" )}";
     DEBUG(3, tiramisu::str_dump("The constraint is:" + constraint2));
 
```

Comment:
the purpose was to devide this computation into 2 computations. the splitting operation didnt take into consideration the case where the iteration didnt start from 0.
That's why we added lower_bound as an additional parameter for our methods.